invenio_oauthclient/contrib/cern.py

```diff
@@ -190,15 +194,38 @@ def account_info(remote, resp):
     """Retrieve remote account information used to find local user."""
     # Query CERN Resources to get user info and groups
     response = remote.get(REMOTE_APP_RESOURCE_API_URL)
+    current_app.logger.error(response.data)
     res = get_dict_from_response(response)
 
     email = res['EmailAddress'][0]
-    common_name = res['CommonName'][0]
+    external_id = res['PersonID'][0]
+    # nice = res['CommonName'][0]
+    # name = res['DisplayName'][0]
 
     # FIXME get user informations accordly with invenio-userprofiles
-    return dict(email=email.lower(), nickname=common_name)
+    return dict(email=email.lower(),
+                external_id=external_id, external_method='cern',
+                active=True, confirmed_at=datetime.utcnow())
 
 
 def account_setup(remote, token, resp):
     """Perform additional setup after user have been logged in."""
-    pass
+    from invenio_oauthclient.utils import oauth_link_external_id
+    from invenio_db import db
+
+    response = remote.get(REMOTE_APP_RESOURCE_API_URL)
+    res = get_dict_from_response(response)
+
+    with db.session.begin_nested():
+        user = token.remote_account.user
+        external_id = res['PersonID'][0]
+
+        # Create user <-> external id link.
+        oauth_link_external_id(user, dict(id=external_id, method="cern"))
+
+    groups = res['Group']
+    for group in groups:
+        g.identity.provides.add(RoleNeed(group))
+
+    identity_changed.send(current_app._get_current_object(),
```

Comment:
either I do that or I call directly `Principal.set_identity(identity)` https://github.com/mattupstate/flask-principal/blob/master/flask_principal.py#L412 (which will have more or less the same effect), otherwise the changes that I'm making to the identity will not be saved to the session and therefore lost